diff --git a/dist/index.js b/dist/index.js
index 2ad1db419fd67377a6ee0af0b76124891fdd6318..2dc0de85ffc625e2fb20a96effebca4c664f7fb1 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -595,7 +595,19 @@ var MentionUI = class extends Plugin {
 				if (data.keyCode == keyCodes.arrowdown) this._mentionsView.selectNext();
 				if (data.keyCode == keyCodes.arrowup) this._mentionsView.selectPrevious();
 				if (commitKeys.includes(data.keyCode)) this._mentionsView.executeSelected();
-				if (data.keyCode == keyCodes.esc) this._hideUIAndRemoveMarker();
+				if (data.keyCode == keyCodes.esc) {
+					this._hideUIAndRemoveMarker();
+
+					editor.model.change(writer => {
+						// insert a zero-width space as a special marker that we don't want a mention active anymore
+						// see e.g. https://github.com/zadam/trilium/issues/4692
+						const insertPosition = editor.model.document.selection.getLastPosition();
+
+						if (insertPosition !== null) {
+							writer.insertText('\u2002', insertPosition);
+						}
+					});
+				}
 			}
 		}, { priority: "highest" });
 		clickOutsideHandler({
@@ -979,9 +991,8 @@ function getLastValidMarkerInText(feedsWithPattern, text) {
 function createRegExp(marker, minimumCharacters) {
 	const numberOfCharacters = minimumCharacters == 0 ? "*" : `{${minimumCharacters},}`;
 	const openAfterCharacters = env.features.isRegExpUnicodePropertySupported ? "\\p{Ps}\\p{Pi}\"'" : "\\(\\[{\"'";
-	const mentionCharacters = ".";
-	marker = marker.replace(/[.*+?^${}()\-|[\]\\]/g, "\\$&");
-	const pattern = `(?:^|[ ${openAfterCharacters}])(${marker})(${mentionCharacters}${numberOfCharacters})$`;
+	const mentionCharacters = "^=\u2002";
+	const pattern = `(?:^|[= ${openAfterCharacters}])([${marker}])([${mentionCharacters}]${numberOfCharacters})$`;
 	return new RegExp(pattern, "u");
 }
 /**
