diff --git a/dist/index.js b/dist/index.js
index 055bdaeb3665fd469b59e09ec1111fc4da20be6c..b405e7c2b76c33e6a41d221389ffff3c2d791300 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -540,6 +540,7 @@ function modelToViewCodeBlockInsertion(model, languageDefs, useLabels = false) {
 		const preAttributes = {};
 		if (useLabels) {
 			preAttributes["data-language"] = languagesToLabels[codeBlockLanguage];
+			preAttributes.class = "hljs";
 			preAttributes.spellcheck = "false";
 		}
 		const codeAttributes = languagesToClasses[codeBlockLanguage] ? { class: languagesToClasses[codeBlockLanguage] } : void 0;
