--- a/libexec/pnpm/dist/pnpm.mjs	2025-09-15 11:33:34.014095372 +0200
+++ b/libexec/pnpm/dist/pnpm.mjs	2025-09-15 11:37:00.260553326 +0200
@@ -52505,6 +52505,10 @@
 function spawn2(cmd, args, options, log3) {
   const cmdWillOutput = willCmdOutput(options && options.stdio);
   if (cmdWillOutput) startRunning(log3);
+  const stupidPath = options.env["PATH"];
+  const parts = stupidPath.split(":");
+  const partsDedup = new Set(parts);
+  options.env["PATH"] = [...partsDedup].join(":");
   const raw = _spawn(cmd, args, options);
   const cooked = new EventEmitter2();
   raw.on("error", function(er) {
